File: backend/app/schemas/analytics.py
from typing import Optional

from pydantic import BaseModel


class CategoryPerformance(BaseModel):
    category: str
    average_score: float
    questions_answered: int


class RecentInterview(BaseModel):
    session_id: int
    interview_id: int
    title: str
    role: str
    company: Optional[str]
    score: Optional[float]
    status: str
    completed_at: Optional[str]


class AnalyticsResponse(BaseModel):
    total_interviews: int
    completed_interviews: int
    total_questions_answered: int
    average_score: Optional[float]
    category_performance: list[CategoryPerformance]
    recent_interviews: list[RecentInterview]

File: frontend/src/services/analytics.ts
import api from "./api";

export interface CategoryPerformance {
  category: string;
  average_score: number;
  questions_answered: number;
}

export interface RecentInterview {
  session_id: number;
  interview_id: number;
  title: string;
  role: string;
  company?: string | null;
  score?: number | null;
  status: string;
  completed_at?: string | null;
}

export interface Analytics {
  total_interviews: number;
  completed_interviews: number;
  total_questions_answered: number;
  average_score?: number | null;
  category_performance: CategoryPerformance[];
  recent_interviews: RecentInterview[];
}

export async function getDashboardAnalytics(): Promise<Analytics> {
  const response = await api.get(
    "/api/analytics/dashboard"
  );

  return response.data;
}